darwin-bootstrap/bundle-loader.c

```diff
@@ -27,7 +27,7 @@ static bool pop(const void **buf, const void *end,
     *buf += sizeof(*op);
     *opc = op->opc;
     if ((size_t) (end - *buf) <= op->namelen ||
-        ((const char *) buf)[op->namelen] != '\0')
+        ((const char *) *buf)[op->namelen] != '\0')
         return false;
```

Comment:
without dereferencing `buf`, you would be checking if the namelen bytes after the pointer was equal to zero rather than the data.